### PR TITLE
FOGL-1035 unit tests for FoglampProcess class

### PR DESCRIPTION
**Pull request by ashwinscale**
Note that register_interest and unregister_interest do not have unit tests as they are currently not implemented. I have opened issue https://scaledb.atlassian.net/browse/FOGL-1119 for this.

### CODE REVIEW

File: python/foglamp/common/process.py [hunk 1]
```diff
@@ -29,6 +29,13 @@ class ArgumentParserError(Exception):
     """ Overwrite default exception to not terminate application """
     pass
 
+class SilentArgParse(argparse.ArgumentParser):
+    def error(self, message):
+        raise ArgumentParserError(message)
+    def silent_arg_parse(self, argument_name):
+        self.add_argument(argument_name)
+        parser_result = self.parse_known_args()
```
Comment (praveen-garg) [inline python/foglamp/common/process.py]: should not we move this to try except? how error def is being used. 

should not we just remove def error! and:

```  try:
          parser_result = parser.parse_known_args()
       except ArgumentParserError:
     raise
       else:
          return list(vars(parser_result[0]).values())[0]
```

File: tests/unit/python/foglamp/common/test_process.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            fp = FoglampProcess()
+        with pytest.raises(TypeError):
+            class FoglampProcessImp(FoglampProcess):
+                pass
+            fp = FoglampProcessImp()
+
+    @pytest.mark.parametrize('argslist',
+                             [([ArgumentParserError()]),
+                              (['corehost', ArgumentParserError()]),
+                              (['corehost', 0, ArgumentParserError()])
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/common/test_process.py]: I don;t think this covers the code path; we should test with  combinations of`address, host and port` itself and for error try some bad say address1

File: python/foglamp/common/process.py (hunk 1 above)
Comment (ashwinscale) [reply python/foglamp/common/process.py]: The regular error method from argparse exits the application if there is an error. We simply want to raise an exception, so I have overwritten the method.

File: tests/unit/python/foglamp/common/test_process.py (hunk 2 above)
Comment (ashwinscale) [reply tests/unit/python/foglamp/common/test_process.py]: I am not sure what you mean. I have mocked an error coming from the silent parser. As we are testing application arguments, and the application, in this case, is the unit test process itself, I chose to mock the parser to simulate actual command line arguments.

File: python/foglamp/common/process.py (hunk 1 above)
Comment (praveen-garg) [reply python/foglamp/common/process.py]: I see, this is overridden method. Please add a comment. thanks for clarification.

File: tests/unit/python/foglamp/common/test_process.py (hunk 2 above)
Comment (praveen-garg) [reply tests/unit/python/foglamp/common/test_process.py]: **Please add test for ValueError handling**

```
[None, None, None] > ValueError: --address is not specified

[None, 1, 'a name'] > ValueError: --address is not specified

['corehost', None, 'a name'] > ValueError: --port is not specified

['corehost', 1, None] > ValueError: --name is not specified
```

I think code should have handling of empty string as well, Not only `None`

e.g. `if self._core_management_host is None or self._core_management_host == "":`

Test data `["", 1, "a name"]`

File: python/foglamp/common/process.py [hunk 3]
```diff
@@ -29,6 +29,13 @@ class ArgumentParserError(Exception):
     """ Overwrite default exception to not terminate application """
     pass
 
+class SilentArgParse(argparse.ArgumentParser):
+    def error(self, message):
```
Comment (praveen-garg) [inline python/foglamp/common/process.py]: please add a comment that is overridden and line break between def blocks

File: tests/unit/python/foglamp/common/test_process.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                              ])
+    def test_constructor_missing_args(self, argslist):
+        class FoglampProcessImp(FoglampProcess):
+            def run():
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/common/test_process.py]: add `self` as first arg as its class's def (for all such occurrences)

File: tests/unit/python/foglamp/common/test_process.py (hunk 2 above)
Comment (ashwinscale) [reply tests/unit/python/foglamp/common/test_process.py]: I didn't add cases for this as I believe the code would never return None. I didn't want to refactor too much as the issue is about adding basic unit tests without refactoring. I have filed an issue for this improvement but I believe it is okay to proceed as is.

**Pull request merged**